### DISCUSSION TITLE
How to make string requires use Roblox hierarchy path instead of file tree path

### DISCUSSION THREAD

**Discussion by ghost** [JohnnyMorganz/luau-lsp#871] [Q&A]
If I want to import something from replicated storage from a script in server script service for example, I can't because it uses file paths relative to the file tree rather than the Roblox hierarchy. Is there a way to fix this?

**Comment by JohnnyMorganz** [comment]
There is currently no way to support this, our implementation of string requires is purely filesystem based at the moment.

We will wait until Luau finalises their string requires set up before making further changes right now.

Roblox has no concepts of string requires at the moment - so you must be using a code transform tool to do this mapping. You might be able to update your tool to map relative to file tree appropriately. Alternatively, you may want to use an alias such as `@ReplicatedStorage` to help resolve this.

**Reply by ghost** [reply]
I don't use any tool to transform string requires to normal requires, string requires already works in studio, although there is no documentation or autocompletion for it yet in studio. 
![image](https://github.com/user-attachments/assets/d06b4075-f57b-43a4-b9bd-fd27a3b80698)

**Reply by JohnnyMorganz** [reply]
String requires in Studio is unstable and subject to change. Unfortunately we don't intend to support this yet until Roblox finalises their solution.
